lib/ansible/modules/cloud/docker/docker_swarm_facts.py

```diff
@@ -0,0 +1,187 @@
+#!/usr/bin/python
+#
+# (c) 2019 Piotr Wojciechowski <piotr@it-playground.pl>
+# (c) Thierry Bouvet (@tbouvet)
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: docker_swarm_facts
+
+short_description: Retrieves facts about Docker Swarm cluster.
+
+description:
+  - Retrieves facts about a Docker Swarm if run on Swarm Manager host.
+  - Retrieves role of node on which it is run.
+
+version_added: "2.8"
+
+author:
+    - Piotr Wojciechowski (@wojciechowskipiotr)
+    - Thierry Bouvet (@tbouvet)
+
+options:
+  list:
+    description:
+      - Returns lists of components.
+      - If C(nodes) the return list of registered nodes (the only supported option now).
+      - If not present then return Swarm facts.
+    choices:
+        - nodes
+    required: false
+  output:
+    description:
+      - If C(short) and I(list) is C(nodes) then returns list of registered nodes names.
+      - If C(long) and I(list) is C(nodes) then returns list as in output of `docker nodes ls`.
+      - Ignored if I(list) is not present.
+    choices:
+      - short
+      - long
+    required: false
+    default: short
+extends_documentation_fragment:
+    - docker
+
+requirements:
+    - "python >= 2.6"
+    - "docker-py >= 1.10.0"
+    - "Please note that the L(docker-py,https://pypi.org/project/docker-py/) Python
+       module has been superseded by L(docker,https://pypi.org/project/docker/)
+       (see L(here,https://github.com/docker/docker-py/issues/1310) for details).
+       For Python 2.6, C(docker-py) must be used. Otherwise, it is recommended to
+       install the C(docker) Python module. Note that both modules should I(not)
+       be installed at the same time. Also note that when both modules are installed
+       and one of them is uninstalled, the other might no longer function and a
+       reinstall of it is required."
+    - "Docker API >= 1.24"
+'''
+
+EXAMPLES = '''
+- name: Get info on Docker Swarm
+  docker_swarm_facts:
+  register: result
+
+- name: Get list of registered nodes
+  docker_swarm_facts:
+    list: nodes
+  register: result
+
+- name: Get extended list of registered nodes
+  docker_swarm_facts:
+    list: nodes
+    output: long
+  register: result
+'''
+
+RETURN = '''
+node_role:
+  description:
+    - Information on role of docker host where M(docker_swarm_facts) is executed.
+    - Will be C(Manager) if executed on Swarm manager.
```

Comment:
Why not use lower-case identifiers? (As you use for option values for `docker_node`, for example?)